application/src/main/java/org/togetherjava/tjbot/commands/help/HelpThreadCommand.java

```diff
@@ -195,9 +197,25 @@ private void closeThread(SlashCommandInteractionEvent event, ThreadChannel helpT
             .setColor(HelpSystemHelper.AMBIENT_COLOR)
             .build();
 
+        manuallyResetChannelActivityCache.invalidate(helpThread);
         event.replyEmbeds(embed).flatMap(any -> helpThread.getManager().setArchived(true)).queue();
     }
 
+    private void resetActivity(SlashCommandInteractionEvent event, ThreadChannel helpThread) {
+        refreshCooldownFor(Subcommand.RESET_ACTIVITY, helpThread);
+
+        if (!event.getMember().equals(helpThread.getOwner())) {
+            event.reply("Only the thread creator can reset activities.").queue();
+            return;
+        }
+
+        helpThread.getHistory().retrievePast(1).queue(messages -> {
+            manuallyResetChannelActivityCache.put(helpThread, messages.get(0).getId());
+            helper.changeChannelActivity(helpThread, HelpSystemHelper.ThreadActivity.LOW);
+            event.reply("Activities have been reset.").queue();
```

Comment:
these lines don't need `messages` so put these out of this lambda

Reply:
done.